File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        current = 0
        _max = 0
        for n in nums:
            if n == 1:
                current += 1
            else:
                _max = max(current, _max)
                current = 0
        
        return max(current, _max)

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-0.rs
impl Solution {
    pub fn length_of_lis(nums: Vec<i32>) -> i32 {
        let mut dp = vec![1; nums.len()];
        let mut longest = 0;

        for i in 0..nums.len() {
            for j in 0..i {
                if nums[j] < nums[i] {
                    dp[i] = dp[i].max(dp[j] + 1)
                }
            }
            longest = longest.max(dp[i])
        }
        longest
    }
}
